### ISSUE TITLE
few questions on geometry angle loss

### ISSUE THREAD

**Issue by krishnakanthnakka** [abduallahmohamed/Social-Implicit#7]
Can you please clarify the implementation of geometric angle losss at [https://github.com/abduallahmohamed/Social-Implicit/blob/b21352990564df10f27acb0ce1e0a18268f8ebc3/train.py#L112](https://github.com/abduallahmohamed/Social-Implicit/blob/b21352990564df10f27acb0ce1e0a18268f8ebc3/train.py#L112)

In the figure provided at [Social-Loss.png](https://github.com/abduallahmohamed/Social-Implicit/blob/b21352990564df10f27acb0ce1e0a18268f8ebc3/images/Social-Loss.png), it is shown angle b/w two edges, however from the code, it looks like the code computes the angle at the origin between any predicted points t and j.  

Can you please clarify further on G-angle loss about its inputs and what angle are we computing? and can you also please tell if V_pred and V_target at [line 88] (https://github.com/abduallahmohamed/Social-Implicit/blob/b21352990564df10f27acb0ce1e0a18268f8ebc3/train.py#L88)
are velocities or absolute predicted points?

**Comment by abduallahmohamed** [comment]
Thanks for asking: 
For the first point: 
It calcs the cosine similarity between two (nodes). Each node location can be seen as a vector and we calc the the cosine angle between both. Your interpretation is correct and the figure should be updated. 

So we are comparing these two angles: 
<img width="875" alt="Screenshot 2023-01-10 at 10 24 27 AM" src="https://user-images.githubusercontent.com/22107547/211632916-41a731bd-b687-46f6-a35f-4ee62e914625.png">


For the second point: 
We used relative velocity with the first point as a normalization method for the trajectories. So in a sense it is velocities.

**Comment by krishnakanthnakka** [comment]
Thank you very much for the clarification and figure.

One last question, if ```V_pred``` and ```V_target``` are velocities,  is the calculation G-distance loss at https://github.com/abduallahmohamed/Social-Implicit/blob/b21352990564df10f27acb0ce1e0a18268f8ebc3/train.py#L112

consistent with the description in the above figure? We are not computing the intra-distance between the predicted location
between $t$ and $j$ but instead dealing with velocities.

I think for ```g-distance loss```, the ```V_pred_min_``` and ```V_target_``` in line 112 should be absolute points instead of velocities. Please correct me if I'm wrong.

**Comment by abduallahmohamed** [comment]
We are dealing with velocities from normalization point of view but originally they are distance. 
The normalization of trajectories can be done by:
1- Relative difference between steps = velocity per each step
2- Divide by a fixe large amount = keep distance
3- Difference with origin point or last point = velocity w.r.t to the origin or last point 

In our case, we do the matching between intra velocities vector, but one can change it based on normalization choice. For example, if I used the abs positions it will explode the gradients. Overall, you are correct in your interpretation but I wanted to highlight the issue of normalization and it is impact on describing the intent of the loss function design.
